c++/src/kj/compat/tls.c++

```diff
@@ -209,33 +209,28 @@ public:
     // Otherwise, certificates issued by Let's Encrypt won't work as of September 30, 2021:
     // https://letsencrypt.org/docs/dst-root-ca-x3-expiration-september-2021/
     X509_VERIFY_PARAM_set_flags(verify, X509_V_FLAG_TRUSTED_FIRST);
+    co_await sslCall([this]() { return SSL_connect(ssl); });
 
-    return sslCall([this]() { return SSL_connect(ssl); }).then([this](size_t) {
-      X509* cert = SSL_get_peer_certificate(ssl);
-      KJ_REQUIRE(cert != nullptr, "TLS peer provided no certificate") { return; }
-      X509_free(cert);
+    X509* cert = SSL_get_peer_certificate(ssl);
+    KJ_REQUIRE(cert != nullptr, "TLS peer provided no certificate") { co_return; }
+    X509_free(cert);
 
-      auto result = SSL_get_verify_result(ssl);
-      if (result != X509_V_OK) {
-        const char* reason = X509_verify_cert_error_string(result);
-        KJ_FAIL_REQUIRE("TLS peer's certificate is not trusted", reason) { break; }
-      }
-    });
+    auto result = SSL_get_verify_result(ssl);
+    if (result != X509_V_OK) {
+      const char* reason = X509_verify_cert_error_string(result);
+      KJ_FAIL_REQUIRE("TLS peer's certificate is not trusted", reason) { break; }
```

Comment:
I know this was here before, but what is that `{ break; }` doing here?

Reply:
Actually, first time I'm seeing this kind of construct. The `co_return` above as well, how does that interact with `KJ_REQUIRE`?